x-pack/legacy/plugins/spaces/public/views/management/edit_space/customize_space/customize_space_avatar.tsx

```diff
@@ -29,9 +42,27 @@ class CustomizeSpaceAvatarUI extends Component<Props, State> {
     super(props);
     this.state = {
       initialsHasFocus: false,
+      avatarImage: props.space.avatarImage,
     };
   }
 
+  private _handleChange = (avImg: string) => {
+    this.setState({ avatarImage: avImg });
+
+    if (avImg) {
+      this.props.onChange({
+        ...this.props.space,
+        avatarImage: avImg,
+      });
+    }
+  };
+
+  private onFileUpload = (files: File[]) => {
+    const [file] = files;
+    const [type, subtype] = get(file, 'type', '').split('/');
+    encode(file).then((dataurl: string) => this._handleChange(dataurl));
```

Comment:
Not a lot of users have access to image applications that quickly allow them to resize their images. I think it'd be a really helpful feature if the Spaces UI could resize the image for them.

Reply:
@legrego : is there an image manipulation library already inside Kibana I can use?
Thanks

Reply:
@legrego : at the end, I used plain javascript.
For size limitation, I used this approach: if the image is below 64x64 pixels, I leave it untouched. Otherwise, I resize the image to stay inside 64xY or Xx64 dimensions.